pkg/providers/aws/cluster_network_provider.go

```diff
@@ -900,3 +911,70 @@ func isValidCIDRRange(CIDR *net.IPNet) bool {
 	mask, _ := CIDR.Mask.Size()
 	return mask > 15 && mask < 27
 }
+
+// standalone vpc cidr block can not overlap with existing cluster vpc cidr block
+// issue arises when trying to peer both vpcs with invalid vpc error - `overlapping CIDR range`
+// this utility function returns true if either cidr range intersect
+func intersect(n1, n2 *net.IPNet) bool {
+	return n2.Contains(n1.IP) || n1.Contains(n2.IP)
+}
+
+// validateStandaloneCidrBlock validates the standalone cidr block before creation, returning an error if the cidr is not valid
+// checks carried out :
+// 		* has a cidr range between \16 and \26
+// 		* does not overlap with cluster vpc cidr block
+func validateStandaloneCidrBlock(ctx context.Context, client client.Client, ec2Api ec2iface.EC2API, logger *logrus.Entry, vpcCidrBlock *net.IPNet) error {
+	// validate has a cidr range between \16 and \26
+	if !isValidCIDRRange(vpcCidrBlock) {
+		return errorUtil.New(fmt.Sprintf("%s is out of range, block sizes must be between `/16` and `/26`, please update `_network` strategy", vpcCidrBlock.String()))
+	}
+
+	clusterVPC, err := getClusterVpc(ctx, client, ec2Api, logger)
+	if err != nil {
+		return errorUtil.Wrap(err, "failed to get cluster vpc")
+	}
+
+	_, clusterVPCCidr, err := net.ParseCIDR(*clusterVPC.CidrBlock)
+	if err != nil {
+		return errorUtil.Wrap(err, "failed to parse cluster vpc cidr block")
+	}
+
+	// validate cidr blocks do not over lap
+	if intersect(vpcCidrBlock, clusterVPCCidr) {
+		return errorUtil.New(fmt.Sprintf("standalone vpc creation failed: standalone cidr block %s overlaps with cluster vpc cidr block %s, update _network strategy to continue vpc creation", vpcCidrBlock.String(), *clusterVPC.CidrBlock))
+	}
+	return nil
+}
+
+// getNetworkProviderConfig return parsed ipNet cidr block
+// a _network resource type strategy, is expected to have the same tier as either postgres or redis resource type
+// ie. for a postgres tier X there should be a corresponding _network tier X
+//
+// the _network strategy config is unmarshalled into a ec2 create vpc input struct
+// from the struct the cidr block is parsed to ensure validity
+func getNetworkProviderConfig(ctx context.Context, configManager ConfigManager, logger *logrus.Entry, tier string) (*net.IPNet, error) {
```

Comment:
Not a big thing, but I feel like the `logger` should be last here, as it's the most "util" of the arguments in the function signature